autogen.pl

```diff
@@ -1085,6 +1085,30 @@ sub patch_autotools_output {
         ;;";
     $c =~ s/$search_string/$replace_string/g;
 
+    push(@verbose_out, $indent_str . "Patching configure for Libtool NVIDIA Fortran compiler name\n");
+    $c =~ s/for ac_prog in gfortran f95 fort xlf95 ifort ifc efc pgfortran pgf95 lf95 f90 xlf90 pgf90 epcf90 nagfor/for ac_prog in gfortran f95 fort xlf95 ifort ifc efc pgfortran pgf95 lf95 f90 xlf90 pgf90 epcf90 nagfor nvfortran/g;
+    foreach my $tag (("", "_FC")) {
+        $search_string = 'tcc\*\)
+	# Fabrice Bellard et al\'s Tiny C Compiler
+	lt_prog_compiler_wl'."${tag}".'=\'-Wl,\'
+	lt_prog_compiler_pic'."${tag}".'=\'-fPIC\'
+	lt_prog_compiler_static'."${tag}".'=\'-static\'
+	;;';
+        $replace_string = "nvfortran*)
+	# NVIDIA Fortran compiler
+	lt_prog_compiler_wl${tag}='-Wl,'
+	lt_prog_compiler_pic${tag}='-fPIC'
+	lt_prog_compiler_static${tag}='-Bstatic'
+	;;
+      tcc*)
+	# Fabrice Bellard et al's Tiny C Compiler
+	lt_prog_compiler_wl='-Wl,'
+	lt_prog_compiler_pic='-fPIC'
+	lt_prog_compiler_static='-static'
+	;;";
```

Comment:
Small suggestion for the `$replace_string`:

```perl
$replace_string = "$search_string
    nvfortran*)
	# NVIDIA Fortran compiler
	lt_prog_compiler_wl${tag}='-Wl,'
	lt_prog_compiler_pic${tag}='-fPIC'
	lt_prog_compiler_static${tag}='-Bstatic'
	;;";
```

That way you don't have to replicate the `tcc*)` block twice.

Reply:
Can't easily because of the protections in `$search_string`.